internal/collector/event.go

```diff
@@ -100,7 +100,15 @@ func beforeLastEvent(t time.Time, ev *v1.Event) bool {
 		return t.Before(ev.Series.LastObservedTime.Time)
 	}
 
-	return t.Before(ev.LastTimestamp.Time)
+	if !ev.LastTimestamp.IsZero() {
```

Comment:
I would like you to add a unit test for this function `beforeLastEvent`, you could easily unit test it with the various edge cases it also might help us find a few other edge cases.

Reply:
@lilic I have added some test scenarios in my last commit, would you mind having another look?